test/variety/dbchoco_test.js

```diff
@@ -0,0 +1,24 @@
+
+// test/variety/dbchoco_test.js
+
+var assert = require('assert');
+
+var pzpr = require('../../');
+
+var puzzle = new pzpr.Puzzle();
+
+describe('Variety:dbchoco',function(){
+	it('Error codes are correct along a series of drawing and undrawing', function(){
+		puzzle.open('dbchoco/2/3/s0l');
+		puzzle.setMode('play');
+		assert.equal(puzzle.check(false)[0], null);
+		puzzle.mouse.inputPath('left', 2,0, 2,2, 2,4, 2,6);
+		assert.equal(puzzle.check(false)[0], 'bkDifferentShape');
+		puzzle.mouse.inputPath('left', 2,0, 2,2, 2,4, 4,4);
```

Comment:
oh I made a mistake here, it's meant to undraw the previous line; should be `2,0, 2,2, 2,4, 2,6` again

Reply:
test fix: https://github.com/robx/pzprjs/commit/4b994593ea745c26864b0a0f880c04b68940bc42

And your latest commit (avoiding graph rebuilds) fixes the test failure, nice.